sites/public/src/lib/listings/search.ts

```diff
@@ -0,0 +1,121 @@
+import { ListingQueryBuilder } from "./listing-query-builder"
```

Comment:
Is there a reason we didn't use a nestjs Data Transfer Object instead of building these client/server query translations?

Reply:
I'm not sure I understand.  DTOs don't have anything to do with query generation, only return values, which the ListingService class does use.

Reply:
Ah ok. I was under the impression that we could use those to pass data between different services.

Reply:
Got it.  The DTOs define the shape of the data controlled by the service, but they don't have any say in how that's queried out.  All the query builder does is construct a set of URL query parameters that get passed to the service so it can then apply filters on the dataset.  These parameters are only loosely related to entity or DTO properties, with other classes on the service side doing the translation between filter keys and database table fields.  For example, when using the `counties` filter, it's actually generating a WHERE condition based on  `building_address->>'county'` since the listing object doesn't actually have a county field.